ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Components/RadioButtonWithStats.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useMemo } from 'react';
+import {
+    Card,
+    CardTitle,
+    CardBody,
+    Radio,
+    FlexItem,
+    Flex,
+    Label,
+    LabelGroup,
+} from '@patternfly/react-core';
+import { BarsIcon, CheckCircleIcon, ExclamationCircleIcon } from '@patternfly/react-icons';
+
+import { ComplianceScanStatsShim } from 'services/ComplianceEnhancedService';
+
+import { getStatusCounts } from '../compliance.coverage.utils';
+
+export type RadioButtonWithStatsProps = {
+    scanStats: ComplianceScanStatsShim;
+    isSelected: boolean;
+    onSelected: (scanName: string) => void;
+};
+
+function RadioButtonWithStats({ scanStats, isSelected, onSelected }: RadioButtonWithStatsProps) {
+    const { scanName } = scanStats;
+    const { passCount, failCount, otherCount } = useMemo(
+        () => getStatusCounts(scanStats.checkStats),
+        [scanStats]
+    );
+
+    const onKeyDown = (event: React.KeyboardEvent) => {
+        if ([' ', 'Enter'].includes(event.key)) {
+            event.preventDefault();
+            onSelected(scanName);
+        }
+    };
+
+    return (
+        <>
+            <Card
+                id={`selectable-card-${scanName}`}
+                onKeyDown={onKeyDown}
+                onClick={() => onSelected(scanName)}
+                hasSelectableInput
+                onSelectableInputChange={() => onSelected(scanName)}
+                isSelectableRaised
+                isSelected={isSelected}
+                isCompact
+            >
+                <CardTitle className="pf-u-p-sm pf-u-pb-0">
+                    <Flex justifyContent={{ default: 'justifyContentFlexEnd' }}>
+                        <FlexItem>
+                            <Radio
+                                id={`selectable-card-radio-${scanName}`}
+                                aria-label={`radio button for ${scanName} coverage`}
+                                name={`selectable-card-radio-${scanName}`}
+                                isChecked={isSelected}
+                            />
+                        </FlexItem>
+                    </Flex>
+                </CardTitle>
+                <CardTitle>{scanName}</CardTitle>
+                <CardBody>
+                    <LabelGroup>
+                        <Label className="pf-u-mr-xs" icon={<CheckCircleIcon />} color="green">
+                            {passCount}
+                        </Label>
+                        <Label className="pf-u-mr-xs" icon={<ExclamationCircleIcon />} color="red">
+                            {failCount}
+                        </Label>
+                        <Label icon={<BarsIcon />} color="grey">
+                            {otherCount}
+                        </Label>
```

Comment:
Each of these labels needs an accessibility name somehow; icon and color are not enough.

This can come in a follow-up PR.

Reply:
added aria labels